cpp/src/parquet/encoding.cc

```diff
@@ -1146,15 +1146,11 @@ int PlainDecoder<DType>::Decode(T* buffer, int max_values) {
   return max_values;
 }
 
-class PlainBooleanDecoder : public DecoderImpl,
-                            virtual public TypedDecoder<BooleanType>,
-                            virtual public BooleanDecoder {
+class PlainBooleanDecoder : public DecoderImpl, virtual public TypedDecoder<BooleanType> {
  public:
   explicit PlainBooleanDecoder(const ColumnDescriptor* descr);
   void SetData(int num_values, const uint8_t* data, int len) override;
 
-  // Two flavors of bool decoding
-  int Decode(uint8_t* buffer, int max_values) override;
```

Comment:
Hi @kou . Why is this public function removed? This breaks the downstream where is depending on it.

Reply:
See https://github.com/apache/arrow/pull/14147#discussion_r977099469 .
Could you share your downstream code that uses this?

@sfc-gh-nthimmegowda Can we keep backward compatibility for this?

Reply:
@kou I believe we can. Setting out a PR later today for this.

Reply:
I have used the template TypedDecoder class which is created by MakeTypedDecoder function defined here: https://github.com/apache/arrow/blob/master/cpp/src/parquet/encoding.h#L447. Then it is casted to BooleanDecoder to use the uint8_t variant.